### PR TITLE
SONARGRADL-113   Migrate to Kotlin DSL

### PR DESCRIPTION
**Pull request by kaufco**

### CODE REVIEW

File: sonarqube-build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            if (artifactoryUsername is String && artifactoryUsername.isNotEmpty() && artifactoryPassword is String && artifactoryPassword.isNotEmpty()) {
+                credentials {
+                    username = artifactoryUsername
+                    password = artifactoryPassword
+                }
+            }
+        }
+    }
+    dependencies {
+        classpath("org.sonarsource.scanner.gradle:sonarqube-gradle-plugin:3.3")
```
Comment (leveretka) [inline sonarqube-build.gradle.kts:27]: should this be the latest version "4.0.0.2929"

File: build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+        !artifactorySkip && (branch == "master" || branch.matches("branch-[\\d.]+".toRegex())) &&
+            gradle.taskGraph.hasTask(":artifactoryPublish")
+    }
+}
+
+tasks.withType<Test>().configureEach {
+    doLast {
+        Thread.sleep(2000) // https://github.com/gradle/gradle/issues/16603
+    }
+}
```
Comment (leveretka) [inline build.gradle.kts:240]: Minor, but there are a few files, that don't have an empty line at the end, would be nice to add it.

**Review by leveretka — COMMENTED** [review]
Looks fine, nice work!

Just a few small comments and please pull the latest changes from the master. For the multiplatform we added a new dependency, that's why you have a conflict.

File: sonarqube-build.gradle.kts (hunk 1 above)
Comment (irina-batinic-sonarsource) [reply sonarqube-build.gradle.kts:27]: I don't see that version here: https://repox.jfrog.io/ui/native/sonarsource/org/sonarsource/scanner/gradle/sonarqube-gradle-plugin/
Am I missing something?
Can we use "4.1.0.3103"?

Comment (leveretka) [reply sonarqube-build.gradle.kts:27]: I'm not sure we can, as 4.0.0.2929 is the latest released and 4.1 is the one we're current;y working on.

Is it failing with 4.0.0.2929?

**Review by leveretka — APPROVED** [review]
Nice work!

**Pull request merged**